cookbooks/aws-parallelcluster-platform/resources/nvidia_imex/partial/_nvidia_imex_common.rb

```diff
@@ -58,11 +60,13 @@
       mode '0644'
       action :create
       variables(imex_main_config_file_path: nvidia_imex_main_conf_file)
+      only_if { Dir.exist?(node['cluster']['nvidia']['imex']['shared_dir']) }
```

Comment:
It's more correct and safer to skip this resource if the service file already exists. In this way we would prevent overwriting the existing file when custom amis are used

Reply:
The service file of IMEX for DLAMI is on a different location. So I am writing the conditions based on or locations instead of finding if their location exists or not as they could change their location in future

Reply:
Added the details in description